packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -83,7 +83,7 @@ interface Signature {
       clearSelections: () => void,
       doWithStableScroll: (changeSizeCallback: () => Promise<void>) => void,
       doScrollIntoView: (selector: string) => void,
-      doCloseAnimation: () => void,
+      doAnimation: (type: 'isClosing' | 'isMovingForward') => void,
```

Comment:
this method has a lot of positional parameters. it's hard to keep track which parameter is for what. usually on the 4th parameter that is added, I switch from positional parameters to named parameters by using an `args` parameter that is an object where each key is one of the positional paramters so that from a readability perspective it is more clear what each parameter is. can we refactor `setupStackItem` to use named parameters instead of positional parameters?

so like this instead:
```ts
setupStackItem: (args: {
      stackItem: StackItem;
      clearSelections: () => void;
      doWithStableScroll: (changeSizeCallback: () => Promise<void>) => void;
      doScrollIntoView: (selector: string) => void;
      doAnimation: (type: 'isClosing' | 'isMovingForward') => void;
}) => void;
```

Reply:
Might even be more elegant to have:

```ts
interface StackItemComponentAPI {  
  clearSelections: () => void;  
  doWithStableScroll: (changeSizeCallback: () => Promise<void>) => void;  
  scrollIntoView: (selector: string) => void;  
  startAnimation: (type: 'isClosing' | 'isMovingForward') => void;  
};

setupStackItem: (model: StackItem, componentAPI: StackItemComponentAPI) => void;
```

Then the OperatorModeStack and InteractSubmode can just keep a `WeakMap<StackItem, StackItemComponentAPI>` which I think is easier to understand than multiple weakmaps with names like `doAnimation`.